new_platforms/docs/new_platform_dev.md

```diff
@@ -1,392 +1,462 @@
-Creating your own enclave and using it from a Windows host app
+Development: Creating Your Own Enclave
 =============
 
-This documentation will walk you through how the calls on the various platforms. This document is a work in progress.
+In this guide, you will learn how to create new applications that take advantage
+of enclaves using the Open Enclave SDK.
 
-## Generating API calls between Trusted Apps ("Enclaves") and Untrusted ("Host") Apps
+# Introduction
 
-To use this SDK, you must define your own
-[EDL](https://software.intel.com/en-us/sgx-sdk-dev-reference-enclave-definition-language-file-syntax)
-file that defines any APIs you want to
-use, and use **oeedger8r** to generate code from it.
-The same generated code will work equally with both SGX
-and OP-TEE, as long as the Untrusted App and the Trusted App both use the right include paths and libs,
-and the following additional code constraint is met:
+Any application that uses the Open Enclave SDK is composed of two parts: the
+host app and the enclave.
 
-OP-TEE only allows one thread per TA to be in an ECALL (i.e., a call into
-a TA from a host app).  Even if it has an OCALL (i.e., an out-call
-back into the host app) in progress, the ECALL must complete before
-another ECALL can enter the TA.  SGX, on the other hand, would allow a
-second ECALL to enter.  So if you want them to function identically, host apps
-can pass the OE\_ENCLAVE\_FLAG\_SERIALIZE\_ECALLS
-flag when creating an enclave to automatically get the OP-TEE like behavior
-for both SGX and TrustZone.
+The interface between the host app and the enclave is defined using the [Enclave
+Definition
+Language](https://software.intel.com/en-us/sgx-sdk-dev-reference-enclave-definition-language-file-syntax),
+or EDL. EDL files describe how a host app calls into an enclave to request a
+secure service (i.e. an Enclave CALL, or ECALL), and how an enclave calls into
+its host app to request an insecure service (i.e. an Out CALL, or OCALL). The
+same EDL file is used to define the interface between the host app and the
+enclave regardless of whether the enclave is an Intel SGX enclave or OP-TEE
+trusted application (TA). An EDL file may include other EDL files and is
+processed using the `oeedger8r` tool. This tool generates two sets of one source
```

Comment:
can't parse grammar in "two sets of one source".  Maybe something missing between "of" and "one"?

Reply:
No, there's more afterwards: "two sets of one source file and one header file", meaning `((interface_t.c, interface_t.h) ; (interface_u.c, interface_u.h))`.